src/lib.rs

```diff
@@ -90,13 +91,18 @@ pub struct ZWaveManager {
 }
 
 impl ZWaveManager {
-    fn new(manager: manager::Manager) -> ZWaveManager {
-        ZWaveManager {
+    fn new(manager: manager::Manager) -> (Self, mpsc::Receiver<ZWaveNotification>) {
+        let (tx, rx) = mpsc::channel();
+
+        let manager = ZWaveManager {
             watcher: ZWaveWatcher {
                 state: Arc::new(Mutex::new(State::new())),
+                sender: Arc::new(Mutex::new(tx))
```

Comment:
I don't understand why you need a mutex here since mpsc is supposed to support multiple senders.

The examples involving multiple threads don't use a mutex either:
https://doc.rust-lang.org/std/sync/mpsc/